### PR TITLE
Apply lots of refinement

### PR DESCRIPTION
**Pull request by seratch**
- Remove StreamEmitter class
- Hide AsyncExecutor from library users, just use ExecutionContext
 - Since it's possible to define thread-pool based ExecutionContext, we don't need to support accepting ExecutorService
- Write lots of scaladocs
- Rename StreamSQL to StreamReadySQL
- Make classes/traits scalikejdbc.streams package private except DatabasePublisher
- Change the packages test classes exist

### CODE REVIEW

File: scalikejdbc-streams/src/main/scala/scalikejdbc/streams/AsyncExecutor.scala [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
+   * Returns current ExecutionContext.
+   */
   def executionContext: ExecutionContext
 
+  /**
+   * Runs an asynchronous operation.
+   */
   def execute(runnable: Runnable): Unit
 
-  def close(): Unit
```
Comment (seratch) [inline scalikejdbc-streams/src/main/scala/scalikejdbc/streams/AsyncExecutor.scala:19]: no need to be Closeable if we drop ExecutorService support.

File: scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabasePublisher.scala [hunk 2]
```diff
@@ -3,44 +3,49 @@ package scalikejdbc.streams
...
- * A database backend Publisher in the context of Reactive Streams
+ * A database backend Publisher in the fashion of Reactive Streams
  *
  * see also: [[http://www.reactive-streams.org/]]
  */
 case class DatabasePublisher[A](
     settings: DatabasePublisherSettings[A],
-    sql: StreamSQL[A],
-    asyncExecutor: AsyncExecutor,
-    streamEmitter: StreamEmitter[A]
```
Comment (seratch) [inline scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabasePublisher.scala:18]: removed StreamEmitter and merged into the Subscription class.

File: scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabasePublisherSettings.scala [hunk 3]
```diff
@@ -5,20 +5,38 @@ import scalikejdbc._
 /**
  * Settings for DatabasePublisher.
  */
-class DatabasePublisherSettings[A](
-  val connectionPoolName: Any,
+private[streams] class DatabasePublisherSettings[A](
+  /**
+   * Connection pool name.
+   */
+  val dbName: Any,
```
Comment (seratch) [inline scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabasePublisherSettings.scala:12]: I suggested to use connectionPoolName but in the point of the view of consistency with existing code, I rethought dbName is also fine. Sorry for bothering you.

File: scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabaseSubscription.scala [hunk 4]
```diff
@@ -405,4 +312,178 @@ private[streams] class DatabaseSubscription[A](
...
+      if (log.isDebugEnabled) {
+        log.debug("Saw transition from demand = 0, but no stream continuation available")
+      }
+    }
+  }
+
+  /**
+   * Emits a bunch of elements.
+   */
+  private[this] def emitDemandedElementsAndReturnRemainingIterator(
```
Comment (seratch) [inline scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabaseSubscription.scala:438]: from StreamEmitter

File: scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabaseSubscription.scala [hunk 5]
```diff
@@ -405,4 +312,178 @@ private[streams] class DatabaseSubscription[A](
...
+    }
+
+    if ((bufferNext && iterator.hasNext) || count == realDemand) iterator
+    else null
+  }
+
+  /**
+   * Cancels a given iterator.
+   */
+  private[this] def cancel(iterator: StreamResultSetIterator[A]): Unit = {
```
Comment (seratch) [inline scalikejdbc-streams/src/main/scala/scalikejdbc/streams/DatabaseSubscription.scala:468]: from StreamEmitter

File: scalikejdbc-streams/src/main/scala/scalikejdbc/streams/package.scala [hunk 6]
```diff
@@ -2,26 +2,31 @@ package scalikejdbc
...
-    sql: StreamSQL[A],
+  /**
+   * Creates a new DatabasePublisher.
+   */
+  private def createDatabasePublisher[A, E <: WithExtractor](
+    sql: StreamReadySQL[A],
     connectionPoolName: Any = ConnectionPool.DEFAULT_NAME
   )(
     implicit
-    asyncExecutor: AsyncExecutor,
```
Comment (seratch) [inline scalikejdbc-streams/src/main/scala/scalikejdbc/streams/package.scala:19]: not to expose AsynExecutor to library users.

**Comment by yoskhdia** [comment]
@seratch 
I fixed bugs:

* https://github.com/scalikejdbc/scalikejdbc/pull/614/commits/0c1c120272fe49cde399a7e57a42f78701d5f830
* https://github.com/scalikejdbc/scalikejdbc/pull/614/commits/d377892b0eeec6d1f4141576128ae7c14ffb8a09

Would you merge these changes and resolve conflicts?

**Comment by seratch** [comment]
@yoskhdia Thanks, I've merged your changes.

**Pull request merged**